packages/terra-date-picker/src/DateRange.jsx

```diff
@@ -0,0 +1,71 @@
+import React, { PropTypes } from 'react';
+import DatePicker from '../lib/DatePicker';
+
+const propTypes = {
+  endDate: PropTypes.oneOfType([PropTypes.object]),
+  startDate: PropTypes.oneOfType([PropTypes.object]),
+  onChange: PropTypes.func,
+};
+
+class DateRange extends React.Component {
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      startDate: props.startDate,
+      endDate: props.endDate,
+    };
+    this.handleChangeStart = this.handleChangeStart.bind(this);
+    this.handleChangeEnd = this.handleChangeEnd.bind(this);
+  }
+
+  handleChange({ startDate, endDate }) {
+    let startDateForRange = startDate || this.state.startDate;
+    let endDateForRange = endDate || this.state.endDate;
+
+    if (startDateForRange.isAfter(endDateForRange)) {
+      const tempDate = startDateForRange;
```

Comment:
This is cool and condense the code to 1 line: https://github.com/cerner/terra-core/commit/c599197256ce3c511ea206ca5e87c4015f4f6fee